content/0007-development-environment/rfc.md

```diff
@@ -0,0 +1,231 @@
+    * Postgresql
+    * Vault
+    * Redis
+    * Ambassador
+* Relay resources are deployed to the cluster using `kubectl apply`.
+    * relay-core CRDs
+* Helm chart configuration values are generated and populated into a
+  `values.yaml` and saved to disk.
+* Relay services are deployed to the cluster using Helm. See how container
+  image management works below.
+    * relay-core
+    * relay-api
+    * relay-ui
+
+#### Image management
+
+The key `rde` being useful is proper container image management. This is a
+feature that allows a developer to easily write code locally and run it against
+a cluster without committing and waiting for a continuous integration cycle.
+
+Container images for Relay services are initially pulled using the `latest` tag
+and retagged to `local` (e.g. `relay-api:local`) unless a `local` tag already
+exists for that image.  These local images are then imported into the cluster by
+`rde`.
+
+When a development build script for a Relay service is ran, the resulting
+docker image is tagged using `local` and if the `rde` utility exists in `$PATH`,
+a call to `rde update images` is made to instruct `rde` to import the new image
+and force a new ReplicaSet to be created for the deployment.
+
+#### Initial user
+
+`rde` has a command `rde user create`, which adds a user record to the Relay
+database.
+
+#### Persisted configuration
+
+In order to start and stop cluster, and manage deployments and users, `rde`
+accesses all the data it needs from a `config.json` file that is populated when
+the cluster is created. This contains the kubeconfig, database credentials,
+vault credentials, helm values, and references to deployment images.
+
+This file is the source of authority for all cluster configuration. Changes made
+to it will reflect down into anything else that exists for the cluster on disk
+(such as the helm values.yaml). This being said, `rde` will not use this file to
+update things like database and vault credentials. Any changes there will need
+to be done manually.
+
+## Engineering-level explanation
+
+While versions beyond V1 have not been planned, I can see requests for changes
+to this system once it's put to use by the team and we should plan for bigger
+changes using extension RFCs to this one. Below I've outlined rough steps that
+should get us to V1.
+
+### V1: Opinionated and just works
+
+The first version should not allow for much configuration. It should provide the
+developer with a working cluster that can update Relay service deployments with
+new container images.
+
+In this version, `rde` will live in `relay-deploy` as a standalone shell script
+or Go wrapper around k3d. This can be installed to a developers `$PATH` either
+by coping the script or binary, or running a `make` target from `relay-deploy`.
+
+This version will require the developer to install a couple hard dependencies in
+order for `rde` to function:
+    * k3d
+    * kubectl
```

Comment:
This bullet list didn't seem to render correctly in the markdown preview. Maybe needs another newline before the list.